core/domain/search_services_test.py

```diff
@@ -229,6 +229,42 @@ def mock_add_docs(docs, index):
 
         self.assertEqual(add_docs_counter.times_called, 1)
 
+    def test_update_public_collection_status_in_search(self):
```

Comment:
What exactly is the behaviour that's being tested here? The action seems to be "update public collection status" but there doesn't seem to be any verification (what are the intended pre- and post-checks)?

Also, you might find that some existing tests do not follow the backend test framework that we laid out in [the wiki](https://github.com/oppia/oppia/wiki/Writing-backend-tests#guidelines-for-writing-good-tests). It is worth updating them accordingly, rather than copying them.

Reply:
Actually the `update_collection_status_in_search` function does not update the collection's current search document as it sends an empty dict as parameter([here](https://github.com/oppia/oppia/blob/develop/core/domain/search_services.py#L311)). Hence the pre and post checks would still yield the same results

Reply:
I see. It looks like there are two cases: one is when the collection is unpublished and the other is when the collection is published.

Perhaps in the former case you can verify that the collection is deleted from the search index. I'm not sure why we have the latter case.

That said, I searched the codebase and it looks like update_collection_status_in_search() is not even used at all. So should we just delete the whole thing? (This is something you should generally check when writing tests.)

Reply:
Removed the function as it was not used anywhere